assets/wizards/engagement/views/newsletters/index.js

```diff
@@ -120,20 +124,119 @@ export const NewspackNewsletters = ( { className, onUpdate, isOnboarding = true
 	);
 };
 
-const Newsletters = () => {
-	const [ { status, url, error, newslettersConfig }, updateConfiguration ] = hooks.useObjectState(
-		{}
+export const SubscriptionLists = ( { onUpdate } ) => {
+	const [ error, setError ] = useState( false );
+	const [ inFlight, setInFlight ] = useState( false );
+	const [ lists, setLists ] = useState( [] );
+	const updateConfig = data => {
+		setLists( data );
+		if ( typeof onUpdate === 'function' ) {
+			onUpdate( data );
+		}
+	};
+	const fetchLists = () => {
+		setError( false );
+		setInFlight( true );
+		apiFetch( {
+			path: '/newspack-newsletters/v1/lists',
+		} )
+			.then( updateConfig )
+			.catch( setError )
+			.finally( () => setInFlight( false ) );
+	};
+	const saveLists = () => {
+		setError( false );
+		setInFlight( true );
+		apiFetch( {
+			path: '/newspack-newsletters/v1/lists',
+			method: 'post',
+			data: { lists },
+		} )
+			.then( updateConfig )
+			.catch( setError )
+			.finally( () => setInFlight( false ) );
+	};
+	const handleChange = ( index, name ) => value => {
+		const newLists = [ ...lists ];
+		newLists[ index ][ name ] = value;
+		updateConfig( newLists );
+	};
+	useEffect( fetchLists, [] );
+	if ( ! lists?.length && ! error ) {
+		return null;
```

Comment:
In principle, I think it'd be better to display some UI (loading or error) at all times. But it's not blocking.